util/db/repository.go

```diff
@@ -216,6 +216,20 @@ func (db *db) DeleteRepository(ctx context.Context, repoURL, project string) err
 	return status.Errorf(codes.NotFound, "repo '%s' not found", repoURL)
 }
 
+func (db *db) GetWriteRepository(ctx context.Context, repoURL string, project string) (*appsv1.Repository, error) {
+	repository, err := db.repoWriteBackend().GetRepository(ctx, repoURL, project)
+	if err != nil {
+		return repository, fmt.Errorf("unable to get write repository %q: %w", repoURL, err)
+	}
+
+	// TODO: enrich with write credentials.
+	//if err := db.enrichCredsToRepo(ctx, repository); err != nil {
+	//	return repository, fmt.Errorf("unable to enrich write repository %q info with credentials: %w", repoURL, err)
+	//}
+
```

Comment:
Forgotten? Not a blocker, but create an issue if it is intended to implement later

Reply:
Added docs, will check for an issue...